delivery_carrier_label_batch/wizard/generate_labels.py

```diff
@@ -86,9 +86,7 @@ def _do_generate_labels(self, group):
                 # generate the label of the pack
                 package_ids = [pack.id] if pack else None
                 try:
-                    picking.with_env(new_env).generate_labels(
-                        package_ids=package_ids
-                    )
+                    picking.with_env(new_env).action_generate_carrier_label()
```

Comment:
@tschanzt This is a performance bottleneck here, before we were generating the label for a given package (`package_ids` above) but now we are generating the labels for all the picking packages in a `for` loop.
I do not know all the code, it needs to be rethought probably.
If you want I can work on it.

Reply:
@sebalix we stil have the threads per picking, so as we discussed in most cases the amount of packages per picking shouldn't be that high and doesn't require further measures.